From edc6512c3e515c6e0143fcff87704d77d6995358 Mon Sep 17 00:00:00 2001
From: Hannes Winkler <hannes.winkler@littlemac.ci.kdab.com>
Date: Wed, 8 Nov 2023 12:04:13 +0100
Subject: [PATCH 1/2] support cross-building linux engine from macos

gn: make linux arm64 build not be auto-detected as a host build
linux embedder: always depend on `libflutter_linux_gtk.so` instead of
`libflutter_${host_os}_gtk.so`
---
 shell/platform/linux/BUILD.gn | 4 ++--
 tools/gn                      | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/shell/platform/linux/BUILD.gn b/shell/platform/linux/BUILD.gn
index b2a8e800f8..608761b8fc 100644
--- a/shell/platform/linux/BUILD.gn
+++ b/shell/platform/linux/BUILD.gn
@@ -303,8 +303,8 @@ zip_bundle("flutter_gtk") {
   }
   tmp_files += [
     {
-      source = "$root_build_dir/libflutter_${host_os}_gtk.so"
-      destination = "libflutter_${host_os}_gtk.so"
+      source = "$root_build_dir/libflutter_linux_gtk.so"
+      destination = "libflutter_linux_gtk.so"
     },
     {
       source = "$root_build_dir/gen_snapshot"
diff --git a/tools/gn b/tools/gn
index 0be76e5477..b4fc590c3a 100755
--- a/tools/gn
+++ b/tools/gn
@@ -97,8 +97,8 @@ def is_host_build(args):
     return True
   # For linux arm64 builds, we cross compile from x64 hosts, so the
   # target_os='linux' and linux-cpu='arm64'
-  if args.target_os == 'linux' and args.linux_cpu == 'arm64':
-    return True
+  # if args.target_os == 'linux' and args.linux_cpu == 'arm64':
+  #   return True
   # The Mac and host targets are redundant. Again, necessary to disambiguate
   # during cross-compilation.
   if args.target_os == 'mac':
@@ -667,7 +667,7 @@ def to_gn_args(args):
   else:
     gn_args['skia_use_gl'] = args.target_os != 'fuchsia'
 
-  if sys.platform == 'darwin' and args.target_os not in ['android', 'fuchsia']:
+  if sys.platform == 'darwin' and args.target_os not in ['android', 'fuchsia', 'linux']:
     # OpenGL is deprecated on macOS > 10.11.
     # This is not necessarily needed but enabling this until we have a way to
     # build a macOS metal only shell and a gl only shell.
-- 
2.41.0.windows.1

